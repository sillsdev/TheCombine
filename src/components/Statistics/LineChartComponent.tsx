import {
  Chart as ChartJS,
  CategoryScale,
  LinearScale,
  Title,
  Tooltip,
  Legend,
  LineElement,
  PointElement,
} from "chart.js";
import distinctColors from "distinct-colors";
import { type ReactElement, useEffect, useState } from "react";
import { Line } from "react-chartjs-2";
import { useTranslation } from "react-i18next";

import { type ChartRootData } from "api/models";

ChartJS.defaults.font.size = 18;
ChartJS.register(
  CategoryScale,
  LinearScale,
  PointElement,
  LineElement,
  Title,
  Tooltip,
  Legend
);

interface LineChartProps {
  isFilterZero?: boolean;
  fetchData: () => Promise<ChartRootData | undefined>;
}

interface DatasetsProps {
  label: string;
  data: Array<number>;
  borderColor?: string;
  backgroundColor?: string;
  tension: 0.4;
  fill: false;
  cubicInterpolationMode: "monotone";
}

interface LineChartDataProps {
  labels: Array<string>;
  datasets: Array<DatasetsProps>;
}

const getDefaultProps = (): LineChartDataProps => ({
  labels: [],
  datasets: [],
});

function FilteredData(numbers: number[]): number[] {
  return numbers.map((num) => (num ? num : NaN));
}

export default function LineChartComponent(
  props: LineChartProps
): ReactElement {
  const [chartData, setChartData] = useState(getDefaultProps());

  useEffect(() => {
    const updateChartList = async (): Promise<void> => {
      const tempData = await props.fetchData();
      const newChartData = getDefaultProps();
      if (tempData) {
        // Get array of unique Color
        const palette = distinctColors({ count: tempData.datasets.length });
        // Update the updateChartData by retrieve
        tempData.dates.map((e) => {
          // trim the format from year-mm-dd to mm-dd
          newChartData.labels.push(e.slice(-5));
        });
        tempData.datasets.forEach((e, index) => {
          newChartData.datasets.push({
            label: e.userName,
            data: props.isFilterZero ? FilteredData(e.data) : e.data,
            borderColor: palette[index].hex().toString(),
            backgroundColor: palette[index].hex().toString(),
            tension: 0.4,
            fill: false,
            cubicInterpolationMode: "monotone",
          });
        });
      }
      setChartData(newChartData);
    };

    updateChartList();
  }, [props]);

  const { t } = useTranslation();
  const chartOptions = {
    responsive: true,
    plugins: { legend: { position: "top" } },
    scales: {
      x: { title: { display: true, text: t("statistics.axisLabel.date") } },
      y: { title: { display: true, text: t("statistics.axisLabel.words") } },
    },
  } as any;

  return <Line data={chartData} options={chartOptions} />;
}
