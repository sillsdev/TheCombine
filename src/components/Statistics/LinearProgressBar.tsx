import Box from "@mui/material/Box";
import LinearProgress from "@mui/material/LinearProgress";
import Typography from "@mui/material/Typography";

interface ProgressBarProps {
  value: number;
}

export default function LinearProgressWithLabel(props: ProgressBarProps) {
  return (
    <Box sx={{ width: "50%" }}>
      <Box sx={{ display: "flex", alignItems: "center" }}>
        <Box sx={{ width: "100%", mr: 1 }}>
          <LinearProgress variant="determinate" {...props} />
        </Box>
        <Box sx={{ minWidth: 35 }}>
          <Typography variant="body2" color="text.secondary">{`${Math.round(
            props.value
          )}%`}</Typography>
        </Box>
      </Box>
    </Box>
  );
}
