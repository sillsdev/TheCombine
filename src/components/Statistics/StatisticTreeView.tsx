import ListSubheader from "@mui/material/ListSubheader";
import List from "@mui/material/List";
import ListItemButton from "@mui/material/ListItemButton";
import ListItemIcon from "@mui/material/ListItemIcon";
import ListItemText from "@mui/material/ListItemText";
import Collapse from "@mui/material/Collapse";
import InboxIcon from "@mui/icons-material/MoveToInbox";
import DraftsIcon from "@mui/icons-material/Drafts";
import SendIcon from "@mui/icons-material/Send";
import ExpandLess from "@mui/icons-material/ExpandLess";
import ExpandMore from "@mui/icons-material/ExpandMore";
import StarBorder from "@mui/icons-material/StarBorder";
import { Card, Grid, makeStyles } from "@material-ui/core";
import { Project, SemanticDomainTimestampNode } from "api";
import { GetSemanticDomainTimestampCounts } from "backend";
import React, { ReactElement, useState, useEffect } from "react";

interface TimeProps {
  currentProject?: Project;
}

const useStyles = makeStyles((theme) => ({
  root: {
    width: "100%",
    maxWidth: "auto",
    backgroundColor: theme.palette.background.paper,
  },
}));

export default function NestedList(props: TimeProps) {
  const [index, setIndex] = useState<string>();
  const classes = useStyles();
  const [list, setList] = React.useState<SemanticDomainTimestampNode[]>([]);

  React.useEffect(() => {
    const updateList = async () => {
      const list = await GetSemanticDomainTimestampCounts(
        props.currentProject!.id
      );
      if (list != undefined) {
        return setList(list);
      }
    };
    updateList();
  }, []);

  function handleList() {
    return list.map((t) => [
      <ListItemButton
        key={t.shortDateString + "time key"}
        onClick={() => setIndex(t.shortDateString)}
        selected={index === t.shortDateString}
      >
        <ListItemText primary={t.shortDateString} />
        {t.count}
      </ListItemButton>,
      <Collapse
        key={t.shortDateString + "collapse"}
        in={index === t.shortDateString}
        timeout="auto"
        unmountOnExit
      >
        {t.nodeList.map((t) => [
          <List
            key={t.shortDateString + "collapse"}
            component="div"
            disablePadding
          >
            <ListItemButton sx={{ pl: 4 }}>
              <ListItemText
                primary={
                  t.hour + "\t--\t" + (t.hour + 1 >= 25 ? 0 : t.hour + 1)
                }
              />
              {t.count}
            </ListItemButton>
          </List>,
        ])}
      </Collapse>,
    ]);
  }

  return (
    <Grid container justifyContent="center">
      <Card style={{ width: 600 }}>
        <List className={classes.root}>{handleList()}</List>
      </Card>
    </Grid>
  );
}
