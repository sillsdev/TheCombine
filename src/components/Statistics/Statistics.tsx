import { Grid, Typography, List, Button } from "@material-ui/core";
import React, { ReactElement, useState, useEffect } from "react";
import { useTranslation } from "react-i18next";

import SemanticDomainStatistics from "./DomainStatistics/SemanticDomainStatistics";
import DomainUserStatistics from "./UserStatistics/DomainUserStatistics";
import { Project } from "api/models";
import { getProject } from "backend";
import * as LocalStorage from "backend/localStorage";
import { defaultWritingSystem } from "types/writingSystem";

export default function Statistics(): ReactElement {
  const [currentProject, setCurrentProject] = useState<Project>();
  const [lang, setLang] = useState<string>(defaultWritingSystem.bcp47);
  const [userView, setUserView] = useState<boolean>(true);
  const { t } = useTranslation();

  useEffect(() => {
    const updateCurrentProject = async () => {
      await getProject(LocalStorage.getProjectId()).then(setCurrentProject);
    };

    updateCurrentProject();
  }, [lang]);

  return (
    <React.Fragment>
      <Grid container direction="row" spacing={1}>
        <Grid item xs={2}>
          <Button
            color="primary"
            variant="contained"
            onClick={() => {
              setUserView(!userView);
            }}
            id={`statistic-button`}
          >
            <Typography variant="h4">
              {userView ? t("statistics.domainView") : t("statistics.userView")}
            </Typography>
          </Button>
        </Grid>

        <Grid
          item
          xs={8}
          container
          direction="column"
          justifyContent="center"
          spacing={2}
        >
          <Grid item>
            <Typography variant="h5" align="center">
              {t("statistics.dataStatistics") + currentProject?.name}
            </Typography>
          </Grid>
          <Grid item>
            <Typography variant="h5" align="center">
              {userView ? t("statistics.userView") : t("statistics.domainView")}
            </Typography>
          </Grid>
          <Grid item>
            {userView && (
              <List>
                <DomainUserStatistics
                  currentProject={currentProject}
                  lang={lang}
                />
              </List>
            )}
          </Grid>
          <Grid item>
            {!userView && (
              <List>
                <SemanticDomainStatistics
                  currentProject={currentProject}
                  lang={lang}
                />
              </List>
            )}
          </Grid>
        </Grid>
      </Grid>
    </React.Fragment>
  );
}
