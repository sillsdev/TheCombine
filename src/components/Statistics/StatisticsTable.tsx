import { Card, Grid, Typography, ListItem, List } from "@material-ui/core";
import React, { ReactElement, useState, useEffect } from "react";

import { Project, SemanticDomain, SemanticDomainTreeNode } from "api/models";
import {
  getAllProjects,
  getAllSemanticDomainTreeNode,
  getAllStatisticsPair,
  getAllWords,
  getProjectName,
} from "backend";
import * as LocalStorage from "backend/localStorage";
import theme from "types/theme";
import { defaultWritingSystem } from "types/writingSystem";

interface StatisticsProps {
  domain: SemanticDomainTreeNode;
  statistics: number;
}

export default function StatisticsTable(props: StatisticsProps): ReactElement {
  return (
    <React.Fragment>
      <Grid container wrap="nowrap" justifyContent="space-around">
        <Grid
          item
          xs={5}
          key={"id_" + props.domain.id}
          style={{
            borderBottomStyle: "dotted",
            borderBottomWidth: 1,
            position: "relative",
          }}
        >
          <Typography variant="body1">{props.domain.id}</Typography>
        </Grid>
        <Grid
          item
          xs={5}
          key={"domain_" + props.domain.id}
          style={{
            borderBottomStyle: "dotted",
            borderBottomWidth: 1,
            position: "relative",
          }}
        >
          <Typography variant="body1">{props.domain.name}</Typography>
        </Grid>
        <Grid
          item
          xs={5}
          key={"statistics_" + props.domain.id}
          style={{
            borderBottomStyle: "dotted",
            borderBottomWidth: 1,
            position: "relative",
          }}
        >
          <Typography variant="body1">{props.statistics}</Typography>
        </Grid>
      </Grid>
    </React.Fragment>
  );
}
