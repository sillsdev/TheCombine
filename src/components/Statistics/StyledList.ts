import { List } from "@mui/material";
import { styled } from "@mui/system";

// https://mui.com/system/styled/#using-the-theme
const StyledList = styled(List)(({ theme }) => ({
  backgroundColor: theme.palette.background.paper,
  maxWidth: "auto",
  width: "100%",
}));

export default StyledList;
