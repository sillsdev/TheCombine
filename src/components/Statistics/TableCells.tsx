import { Grid, Typography } from "@mui/material";
import { ReactElement } from "react";
import { useTranslation } from "react-i18next";

export function ColumnHead(props: { titleId: string }): ReactElement {
  const { t } = useTranslation();
  return (
    <Grid
      item
      xs={5}
      style={{
        borderBottomStyle: "dotted",
        borderBottomWidth: 1,
        position: "relative",
      }}
    >
      <Typography variant="subtitle1">{t(props.titleId)}</Typography>
    </Grid>
  );
}

export function TableCell(props: {
  text?: string | number | null;
}): ReactElement {
  return (
    <Grid
      item
      xs={5}
      style={{
        borderBottomStyle: "dotted",
        borderBottomWidth: 1,
        position: "relative",
      }}
    >
      <Typography variant="body1">{props.text}</Typography>
    </Grid>
  );
}
