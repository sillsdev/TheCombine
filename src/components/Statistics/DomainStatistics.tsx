import { Card, Grid, ListItem, List } from "@mui/material";
import { ReactElement, useState, useEffect } from "react";

import { SemanticDomainCount, SemanticDomainTreeNode } from "api/models";
import { getSemanticDomainCounts } from "backend";
import * as LocalStorage from "backend/localStorage";
import { ColumnHead, TableCell } from "components/Statistics/TableCells";

interface DomainStatisticsProps {
  lang: string;
}

export default function DomainStatistics(
  props: DomainStatisticsProps
): ReactElement {
  const [statisticsList, setStatisticsList] = useState<SemanticDomainCount[]>(
    []
  );

  useEffect(() => {
    const updateStatisticList = async (): Promise<void> => {
      const counts = await getStatisticsCounts(
        LocalStorage.getProjectId(),
        props.lang
      );
      if (counts !== undefined) {
        return setStatisticsList(counts);
      }
    };
    updateStatisticList();
  }, [props.lang]);

  async function getStatisticsCounts(
    projectId: string,
    lang?: string
  ): Promise<SemanticDomainCount[] | undefined> {
    return await getSemanticDomainCounts(projectId, lang);
  }

  return (
    <Grid container justifyContent="center">
      <Card style={{ width: 600 }}>
        <List>
          <Grid container wrap="nowrap" justifyContent="space-around">
            <ColumnHead titleId={"statistics.column.domainNumber"} />
            <ColumnHead titleId={"statistics.column.domainName"} />
            <ColumnHead titleId={"statistics.column.senseCount"} />
          </Grid>
        </List>
        <List>
          {statisticsList.map((t) => (
            <TableRow
              key={t.semanticDomainTreeNode.id}
              dom={t.semanticDomainTreeNode}
              count={t.count}
            />
          ))}
        </List>
      </Card>
    </Grid>
  );
}

function TableRow(props: {
  dom: SemanticDomainTreeNode;
  count: number;
}): ReactElement {
  return (
    <ListItem style={{ minWidth: "600px" }}>
      <Grid container wrap="nowrap" justifyContent="space-around">
        <TableCell text={props.dom.id} />
        <TableCell text={props.dom.name} />
        <TableCell text={props.count} />
      </Grid>
    </ListItem>
  );
}
