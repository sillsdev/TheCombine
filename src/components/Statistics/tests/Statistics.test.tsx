import renderer, {
  ReactTestInstance,
  ReactTestRenderer,
} from "react-test-renderer";

import "tests/mockReactI18next";

import Statistics from "components/Statistics/Statistics";
import { newProject } from "types/project";

let testRenderer: ReactTestRenderer;

const mockProject = newProject();
const mockProjectId = "mockProjectId";

const mockGetProject = jest.fn();
const mockGetProjectId = jest.fn();
const mockGetDomainUserCounts = jest.fn();
const mockGetSemanticDomainCounts = jest.fn();

jest.mock("backend", () => ({
  getProject: (projectId: string) => mockGetProject(projectId),
  getSemanticDomainUserCount: () => mockGetDomainUserCounts(),
  getSemanticDomainCounts: () => mockGetSemanticDomainCounts(),
}));

jest.mock("backend/localStorage", () => ({
  getProjectId: () => mockGetProjectId(),
}));

function setMockFunctions() {
  mockGetProjectId.mockReturnValue(mockProjectId);
  mockGetProject.mockResolvedValue(mockProject);
}

beforeEach(async () => {
  jest.clearAllMocks();
  setMockFunctions();
  await renderer.act(async () => {
    testRenderer = renderer.create(<Statistics />);
  });
});

describe("Statistics", () => {
  it("renders without crashing, UI does not change unexpectedly", async () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
  it("useEffect hook was called", async () => {
    //Verify the mock function called
    expect(mockGetProject).toBeCalled();
    expect(mockGetProjectId).toBeCalled();
  });
});
