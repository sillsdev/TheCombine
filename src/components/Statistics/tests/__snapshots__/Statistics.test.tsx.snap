// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Statistics renders without crashing, UI does not change unexpectedly 1`] = `
Array [
  <div
    className="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-1 css-a5rdam-MuiGrid-root"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
    >
      <ul
        className="MuiList-root MuiList-padding makeStyles-root-1 css-h4y409-MuiList-root"
      >
        <div
          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters Mui-selected MuiListItemButton-root MuiListItemButton-gutters Mui-selected css-1l1zyj1-MuiButtonBase-root-MuiListItemButton-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-w8l8fw-MuiTypography-root"
            >
              statistics.userView
            </span>
          </div>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1l1zyj1-MuiButtonBase-root-MuiListItemButton-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-w8l8fw-MuiTypography-root"
            >
              statistics.domainView
            </span>
          </div>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1l1zyj1-MuiButtonBase-root-MuiListItemButton-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-w8l8fw-MuiTypography-root"
            >
              statistics.dayView
            </span>
          </div>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <div
          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-1l1zyj1-MuiButtonBase-root-MuiListItemButton-root"
          onBlur={[Function]}
          onClick={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={0}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-w8l8fw-MuiTypography-root"
            >
              statistics.estimate
            </span>
          </div>
          <span
            className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </div>
      </ul>
    </div>
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-2 MuiGrid-direction-xs-column MuiGrid-grid-xs-8 css-l4fqoi-MuiGrid-root"
    >
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1pb9bi6-MuiTypography-root"
        >
          statistics.dataStatistics
        </h5>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <h5
          className="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-1pb9bi6-MuiTypography-root"
        >
          statistics.userView
        </h5>
      </div>
      <div
        className="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
      >
        <ul
          className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <div
            className="MuiGrid-root MuiGrid-container css-1lym95h-MuiGrid-root"
          >
            <div
              className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
              style={
                Object {
                  "width": 600,
                }
              }
            >
              <ul
                className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              >
                <div
                  className="MuiGrid-root MuiGrid-container MuiGrid-wrap-xs-nowrap css-1ymxtd0-MuiGrid-root"
                >
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                    style={
                      Object {
                        "borderBottomStyle": "dotted",
                        "borderBottomWidth": 1,
                        "position": "relative",
                      }
                    }
                  >
                    <h6
                      className="MuiTypography-root MuiTypography-subtitle1 css-1m0pdmy-MuiTypography-root"
                    >
                      statistics.userName
                    </h6>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                    style={
                      Object {
                        "borderBottomStyle": "dotted",
                        "borderBottomWidth": 1,
                        "position": "relative",
                      }
                    }
                  >
                    <h6
                      className="MuiTypography-root MuiTypography-subtitle1 css-1m0pdmy-MuiTypography-root"
                    >
                      statistics.domainCount
                    </h6>
                  </div>
                  <div
                    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-5 css-k3f0g4-MuiGrid-root"
                    style={
                      Object {
                        "borderBottomStyle": "dotted",
                        "borderBottomWidth": 1,
                        "position": "relative",
                      }
                    }
                  >
                    <h6
                      className="MuiTypography-root MuiTypography-subtitle1 css-1m0pdmy-MuiTypography-root"
                    >
                      statistics.senseCount
                    </h6>
                  </div>
                </div>
              </ul>
              <ul
                className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
              />
            </div>
          </div>
        </ul>
      </div>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-2 css-1o7apob-MuiGrid-root"
    >
      <ul
        className="MuiList-root MuiList-padding makeStyles-root-2 css-h4y409-MuiList-root"
      >
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1i8meaj-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-w8l8fw-MuiTypography-root"
            >
              statistics.domainProgress
            </span>
          </div>
          <div
            className="MuiBox-root css-19uqgym"
          >
            <div
              className="MuiBox-root css-1bhivts"
            >
              <span
                aria-valuemax={100}
                aria-valuemin={0}
                aria-valuenow={0}
                className="MuiLinearProgress-root MuiLinearProgress-colorPrimary MuiLinearProgress-determinate css-l22418-MuiLinearProgress-root"
                role="progressbar"
              >
                <span
                  className="MuiLinearProgress-bar MuiLinearProgress-barColorPrimary MuiLinearProgress-bar1Determinate css-worpmc-MuiLinearProgress-bar1"
                  style={
                    Object {
                      "transform": "translateX(-100%)",
                    }
                  }
                />
              </span>
            </div>
            <div
              className="MuiBox-root css-0"
            >
              <p
                className="MuiTypography-root MuiTypography-body2 css-1svc9sv-MuiTypography-root"
              >
                0%
              </p>
            </div>
          </div>
        </li>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1i8meaj-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-w8l8fw-MuiTypography-root"
            >
              statistics.domainsCollected
            </span>
          </div>
          <p
            className="MuiTypography-root MuiTypography-body1 css-15ouo8w-MuiTypography-root"
          >
            0
          </p>
        </li>
        <hr
          className="MuiDivider-root MuiDivider-fullWidth css-9mgopn-MuiDivider-root"
        />
        <li
          className="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1i8meaj-MuiListItem-root"
          disabled={false}
        >
          <div
            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
          >
            <span
              className="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-w8l8fw-MuiTypography-root"
            >
              statistics.wordsCollected
            </span>
          </div>
          <p
            className="MuiTypography-root MuiTypography-body1 css-15ouo8w-MuiTypography-root"
          >
            0
          </p>
        </li>
      </ul>
    </div>
  </div>,
  " ",
]
`;
